### PR TITLE
Refactor `lj-union`.

### PR DESCRIPTION
**Pull request by joaander**
## Description

Port improvements made in #56 to `lj-union`.

## Motivation and context

Make consistent plots, reuse code in util, and allow `lj-union` to be run on the CPU only.

## How has this been tested?

I executed the `lj-union` tests on Bridges-2 and Delta. Results: 
lj_union_compare_kT1 5_density0 04
lj_union_conservation_kT1 5_density0 04_N512
lj_union_distribution_analyze_kT1 5_density0 04

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-validation/blob/trunk/CONTRIBUTING.md).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-validation/blob/trunk/ContributorAgreement.md).
- [x] My name is on the list of contributors (`CONTRIBUTORS.md`) in the pull request source branch.

### CODE REVIEW

File: hoomd_validation/util.py [hunk 1]
```diff
@@ -266,11 +266,16 @@ def plot_vs_expected(ax,
...
+    avg_value = {}
+    stderr_value = {}
+    for mode in sim_modes:
+        if numpy.all(numpy.isnan(values[mode])):
+            avg_value[mode] = numpy.nan
+            stderr_value[mode] = numpy.nan
+        else:
+            avg_value[mode] = numpy.mean(values[mode])
+            stderr_value[mode] = 2 * numpy.std(values[mode]) / numpy.sqrt(
+                len(values[mode]))
```
Comment (b-butler) [inline hoomd_validation/util.py:278]: These will be nan if any value is none. Is that the desired behavior? If so the change is unnecessary. If it isn't the desired behavior then `nanmean` and `nanstd` are likely what you want.

File: hoomd_validation/lj_union.py [hunk 2]
```diff
@@ -770,191 +899,112 @@ def lj_union_analyze(job):
...
+        if 'md' in sim_mode:
+            pressures[sim_mode] = log_traj[
+                'log/md/compute/ThermodynamicQuantities/pressure']
         else:
-            densities[sim_mode] = numpy.ones(len(
-                energies[sim_mode])) * job.statepoint.density
+            pressures[sim_mode] = numpy.full(len(energies[sim_mode]), numpy.nan)
+
+        densities[sim_mode] = log_traj[
+            'log/custom_actions/ComputeDensity/density']
```
Comment (b-butler) [inline hoomd_validation/lj_union.py:958]: This could be combined with the previous conditional

**Review by b-butler — COMMENTED** [review]
I just have one comment to ensure the behavior is what you desire. Otherwise looks ready to merge.

File: hoomd_validation/util.py (hunk 1 above)
Comment (joaander) [reply hoomd_validation/util.py:278]: Yes, this is working as intended. For a given mode (e.g. `npt_mc_gpu`) either values will be all real values (e.g. `density`) or they will all be `nan` (e.g. `pressure`). We will never have some replicates with real values and some with nan.

Below on line 301, I use `nanmean` when averaging across modes where a given value (e.g. `pressure`) will be nan for some modes (MC) and real for others (MD).

**Review by b-butler — APPROVED** [review]

**Pull request merged**